.github/workflows/dev-test.yml

```diff
@@ -0,0 +1,140 @@
+# Run tests against developer push
+
+# Runs jobs on linux and windows to maintain compatibility.
+# The main repo is checked out into a sub-directory because an optional
+# second repo may be checked out for testing.
+
+name: dev-test
+
+on:
+  push:
+    branches-ignore: [ master, main ]
+
+jobs:
+  build-ubu:
+    runs-on: ubuntu-latest
+
+    # job level env vars used for a couple of steps
+    env:
+      # see github-glue.test.ts for info on setting these
+      GGG_TOKEN: ${{ secrets.GGG_TOKEN }}
+      GGG_REPOSITORY: ${{ secrets.GGG_REPOSITORY }}
+
+    steps:
+    # Check out repo under sub-dir so other repo can be checked out
+    - uses: actions/checkout@v2
+      with:
+        path: gitgitgadget
+
+    # Check out github-glue.test.ts repo if configured for it
+    - uses: actions/checkout@v2
+      with:
+        repository: ${{ github.repository_owner }}/${{ env.GGG_REPOSITORY }}
+        token: ${{ env.GGG_TOKEN }}
+        path: ${{ env.GGG_REPOSITORY }}
+      if: env.GGG_TOKEN
```

Comment:
Hmm. I'd rather move that repository to the `gitgitgadget` org, then make this step contingent on `${{ github.repository_owner }} == 'gitgitgadget'`  and clone it always, using a shell step if necessary.

Reply:
Oh! I think I understand now. This is a fork of `gitgitgadget/git` for testing purposes?

Reply:
It is a repo for testing purposes.  It does not have to be a fork.  I have [webstech/gitout](https://github.com/webstech/gitout) set up for this.  The github-glue.test.ts uses it to open a PR, create comments, etc.  Since it can be a private repo, a PAT is needed.  A new github repo could be created/deleted but using an existing one was easier and faster.  gitgitgadget would need one for CI tests if we don't add the create ability.

Reply:
That's good! Maybe we can add a YAML comment at the top to explain this?

Reply:
> That's good! Maybe we can add a YAML comment at the top to explain this?

Renamed and comments added.